### DISCUSSION TITLE
Welcome to vscode-stationeers-ic10 Discussions!

### DISCUSSION THREAD

**Discussion by Traineratwot** [Stationeers-ic/vscode-stationeers-ic10#35] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by joshuareeves** [comment]
Thanks for putting this together! It's made a huge difference being able to work through my code using a modern editor instead of the in-game editor.

**Comment by enochwu** [comment]
Just wanna say, you guys rock! Sure beats doing this in-game. Keep up the hard work, I may contribute in the future if there's anything, but so far, it looks like you guys have this linter/extension down. 8)

**Comment by mariokoehler** [comment]
i was wondering if using parenthesis when evaluating boolean expressions in if statements is something you would consider as a future feature?

So, for example an expression like "a && (b || c)". currently you must write it like this to get the correct result

```
if b || c && a
 ...
end
```

but with more complex expressions it's not always possible to have these evaluated in a single if statement (e.g. "(a && b) || (c && d)" can't be rewritten to the current icx syntax in a single line, as far as i'm aware) , you might have to prepare sub-results in helper variables. So it would be great if we could write the syntax like this and have the compiler do the work:

```
if (a && b) || (c && d)
 ...
end
```

i realize it's not a trivial feature, especially with nested parenthesis. maybe you've already considered it and decided against it, but it would be interesting to hear your take on it.

to be honest, before ICX/IC10 i never gave much thought about what the compiler has to do to translate a high level programming language into assembler. I just took it for granted that it works. and lets face it i'm probably too dumb anyway to understand in detail how a C or C++ compiler works :-) 

But thinking about this stuff on the much easier level that is ICX/IC10 it made me appreciate my Java compiler/interpreter even more :-)

**Reply by Traineratwot** [reply]
I've never done compilers :) I am a web developer. I thought about the conditions with brackets, but I didn't come up with anything useful, brackets are very difficult. They work in mathematical expressions, but only because I found a good library

**Reply by mariokoehler** [reply]
ah, i see. i just assumed you used available compiler generator tools. it seems like a perfect usecase for it. but i wouldn't know where to start either, i've never done compilers either. might be a good usecase to start, though :-) aah, so many interesting things to do, so little time.

but then again, ICX is in a good place, it's a really helpful tool. i enjoy using it. thanks for your efforts!

**Comment by nishanh** [comment]
Great extension! It doesn't understand that you can "alias an alias" though:

alias rCurrentTemp r0
alias rCompare1 rCurrentTemp

Syntax highlighter flags the second line as 'must be R/D'

This is valid syntax in-game. The script runs :) I'm doing this as an effective register swap later in the script.

**Reply by Traineratwot** [reply]
it's a funny opportunity, but I won't implement it, I can't even imagine why a link to a link is needed

**Comment by Grayzer** [comment]
а можно починить в компиляторе выражение типа
`while nCounter <> tCounter`
`....`
`end`
а то в ic10 вместо
`while0:`
`bne nCounter tCouner while0exit`
пишет
`while0:`
`beqz undefined while0exit`
?

и немного нехватает select'а ))

**Reply by Traineratwot** [reply]
да, это проблема, я исправлю ее сегодня или завтра

**Reply by Traineratwot** [reply]
готово

**Comment by WernerLuiz92** [comment]
Man, what a brilliant idea!!! Congratulations on your work. I'm a software developer and my hobby after work is playing stationeers and inventing crazy things, just like a mad scientist lol. I was thinking about trying to create something like a python interpreter for IC10, but I don't need to do that anymore. Thank you and, if you need any help with your work, if it's within my knowledge, you can count on me!

**Reply by Traineratwot** [reply]
Come to our discord :) 
https://discord.gg/hcdFGBBX

And try the new version of the interpreter
https://ic10.dev/

Currently, icX does not support polygamy of new features in Game. and new wersion in development
